sts-mock/src/token/keyRetriever/tests/keyRetriever.test.ts

```diff
@@ -0,0 +1,80 @@
+import {
+  GetObjectCommand,
+  GetObjectCommandOutput,
+  S3Client,
+} from "@aws-sdk/client-s3";
+import { mockClient } from "aws-sdk-client-mock";
+import { KeyRetriever, SigningKey } from "../keyRetriever";
+
+const mockS3Client = mockClient(S3Client);
+const mockS3Response = (content: string) => {
+  return {
+    transformToString: async () => content,
+  };
+};
+
+describe("Key Retriever", () => {
```

Comment:
reminder - add a test for the scenario where the body is undefined in the S3 response :)

Reply:
Done - see line 46 of this file